src/Core/Result.php

```diff
@@ -139,6 +148,17 @@ public function __destruct()
         }
 
         $this->response->cancel();
+        unset($this->response);
```

Comment:
Why do we need to unset the response? The `Result` object is being destructed.

Reply:
if we don't, the Result's destructor calls `$this->resolve()` which triggers a NetworkException because the `$response` is already canceled

Reply:
Oh, yes. that is true. 

Thank you.